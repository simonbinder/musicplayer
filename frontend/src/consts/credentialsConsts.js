export const SAVE_SPOTIFY_ACCESS_TOKEN = "credentials/save-spotify-access-token";
export const SAVE_USER = "credentials/save-user";
export const RESET_CREDENTIALS = "credentials/reset-credentials";
