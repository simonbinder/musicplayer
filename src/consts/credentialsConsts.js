export const SAVE_SPOTIFY_ACCESS_TOKEN = "credentials/save-spotify-access-token";
