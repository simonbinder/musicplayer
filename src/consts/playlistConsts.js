export const SAVE_PLAYLISTS_INITIAL = "playlist/save-playlists-initial";
export const SAVE_PLAYLIST = "playlist/save-new-playlist";
export const REMOVE_PLAYLIST = "playlist/remove-playlist";
export const UPDATE_PLAYLIST = "playlist/update-playlist";
export const PLAY_TRACK = "playlist/play-track";
export const SHUFFLE_STATE_CHANGED = "playlist/shuffle-state-changed";
