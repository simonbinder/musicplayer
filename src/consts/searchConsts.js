export const SEARCH_VALUE_CHANGED = "search/search-value-changed";
