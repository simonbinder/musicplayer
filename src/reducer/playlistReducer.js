import {
  SAVE_PLAYLISTS_INITIAL,
  SAVE_PLAYLIST,
  REMOVE_PLAYLIST,
  PLAY_TRACK,
  SHUFFLE_STATE_CHANGED,
  UPDATE_PLAYLIST
} from '../consts/playlistConsts';

const initialState = {
  playStatus: 'paused',
  activeTrack: null,
  playlists: [],
  shuffle: false,
};

export default function playlistReducer(state = initialState, action) {
  switch(action.type) {
    case SAVE_PLAYLISTS_INITIAL:
      return Object.assign({}, state, {
        playlists: action.payload,
      });
    case SAVE_PLAYLIST:
      return Object.assign({}, state, {
        playlists: state.playlists.concat(action.payload),
      });
    case REMOVE_PLAYLIST:
      return Object.assign({}, state, {
        playlists: state.playlists.filter(playlist => {
          return playlist._id != action.payload; //its an id
        }),
      });
    case PLAY_TRACK:
      return Object.assign({}, state, {
        activeTrack: action.payload,
      });
    case SHUFFLE_STATE_CHANGED:
      return Object.assign({}, state, {
        shuffle: !state.shuffle,
      });
    case UPDATE_PLAYLIST:
      let index = state.playlists.findIndex(item => item._id == action.payload._id);

      if(index == -1) {
        return state;
      }

      return Object.assign({}, state, {
        playlists: state.playlists.map((playlist, i) => {
          if(i == index) {
            return action.payload;
          }

          return playlist;
        })
      });
    default:
      return state;
  }
};
