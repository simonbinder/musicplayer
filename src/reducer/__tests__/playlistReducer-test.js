import playlistReducer from '../playlistReducer'
import {SAVE_PLAYLISTS_INITIAL, SAVE_PLAYLIST, REMOVE_PLAYLIST, PLAY_TRACK} from '../../consts/playlistConsts';

describe('search reducer', () => {
  it('should return the initial state', () => {
    expect(playlistReducer(undefined, {})).toMatchSnapshot();
  })

  it('save initial playlist', () => {
    expect(playlistReducer({}, {
      type: SAVE_PLAYLISTS_INITIAL,
      payload: "Test"
    })).toMatchSnapshot();
  })

  it('play track', () => {
    expect(playlistReducer({}, {
      type: PLAY_TRACK,
      payload: "Test"
    })).toMatchSnapshot();
  })

  it('save playlist', () => {
    const initialState = {
      activeTrack: null,
      playlists: "test"
    };
    expect(playlistReducer(initialState, {
      type: SAVE_PLAYLIST,
      payload: "Test"
    })).toMatchSnapshot();
  })

  it('remove playlist', () => {
    const initialState = {
      activeTrack: null,
      playlists: [
        {
          _id: 1,
          name: "test"
        }, {
          _id: 2,
          name: "test2"
        }
      ]

    };
    expect(playlistReducer(initialState, {
      type: REMOVE_PLAYLIST,
      payload: 2
    })).toMatchSnapshot();
  })
})
