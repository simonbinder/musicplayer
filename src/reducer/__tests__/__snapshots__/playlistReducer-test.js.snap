// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search reducer play track 1`] = `
Object {
  "activeTrack": "Test",
}
`;

exports[`search reducer remove playlist 1`] = `
Object {
  "activeTrack": null,
  "playlists": Array [
    Object {
      "_id": 1,
      "name": "test",
    },
  ],
}
`;

exports[`search reducer save initial playlist 1`] = `
Object {
  "playlists": "Test",
}
`;

exports[`search reducer save playlist 1`] = `
Object {
  "activeTrack": null,
  "playlists": "testTest",
}
`;

exports[`search reducer should return the initial state 1`] = `
Object {
  "activeTrack": null,
  "playStatus": "paused",
  "playlists": Array [],
}
`;
